ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -158,37 +178,90 @@ func (rdwr *YAMLReadWriter) Get(clusterConfig string, profileConfig string) (*CL
 
 }
 
-// Save saves the CLIConfig to a yaml formatted file
-func (rdwr *YAMLReadWriter) Save(cliConfig *CLIConfig) error {
-	destMode := rdwr.destination.Mode
-	if err := os.MkdirAll(rdwr.destination.Path, *destMode); err != nil {
-		return errors.Wrapf(err, "Could not make directory %s", rdwr.destination.Path)
+func (rdwr YAMLReadWriter) saveConfig(path string, config interface{}) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return errors.Wrap(err, "Error saving config file")
 	}
 
-	path := configFilePath(rdwr.destination)
+	destMode := rdwr.destination.Mode
+	err = os.MkdirAll(rdwr.destination.Path, *destMode)
+	if err != nil {
+		return err
+	}
 
 	// If config file exists, set permissions first, because we may be writing creds.
-	// This is necessary because ioutil.WriteFile only sets the permissions
-	// if the file is being created for the first time; this handles the case
-	// where the file already exists
 	if _, err := os.Stat(path); err == nil {
 		if err = os.Chmod(path, configFileMode); err != nil {
-			return errors.Wrapf(err, "Could not set file permissions, %s, for path %s", configFileMode, path)
+			logrus.Errorf("Unable to chmod %s to mode %s", path, configFileMode)
+			return err
 		}
 	}
 
-	data, err := yaml.Marshal(cliConfig)
+	err = ioutil.WriteFile(path, data, configFileMode.Perm())
 	if err != nil {
-		return errors.Wrap(err, "Unable to parse the config")
-	}
-	// WriteFile will not change the permissions of an existing file
-	if err = ioutil.WriteFile(path, data, configFileMode.Perm()); err != nil {
-		return errors.Wrapf(err, "Could not write config file %s", path)
+		logrus.Errorf("Unable to write config to %s", path)
+		return err
 	}
 
 	return nil
 }
 
+// SaveProfile saves a single credential configuration
+func (rdwr YAMLReadWriter) SaveProfile(configName string, profile *Profile) error {
+	path := credentialsFilePath(rdwr.destination)
+	config, err := readCredFile(path)
+	if err != nil {
+		return err
+	}
+
+	config.Profiles[configName] = *profile
+
+	// save the modified config
+	return rdwr.saveConfig(path, config)
+}
+
+// SaveCluster save a single cluster configuration
+func (rdwr YAMLReadWriter) SaveCluster(configName string, cluster *Cluster) error {
+	path := configFilePath(rdwr.destination)
+	config, err := readClusterFile(path)
+	if err != nil {
+		return err
+	}
+
+	config.Clusters[configName] = *cluster
+
+	// save the modified config
+	return rdwr.saveConfig(path, config)
+}
+
+// SetDefaultProfile updates which set of credentials is defined as default
+func (rdwr YAMLReadWriter) SetDefaultProfile(configName string) error {
+	path := credentialsFilePath(rdwr.destination)
+	config, err := readCredFile(path)
+	if err != nil {
+		return err
+	}
+
+	config.Default = configName
```

Comment:
Hmm. It would eventually result in an error. But I think I'll add a check to make sure `configName` is a real configuration.